lib/rucio/daemons/automatix/automatix.py

```diff
@@ -239,23 +241,32 @@ def run_once(heartbeat_handler: HeartbeatHandler, inputfile: str, **_kwargs) ->
                     file_["dataset_meta"]["lifetime"] = dataset_lifetime
             files.append(file_)
         logger(logging.INFO, "Upload %s:%s to %s", scope, dsn, rse)
-        upload_client = UploadClient(client)
-        ret = upload_client.upload(files)
-        if ret == 0:
-            logger(logging.INFO, "%s successfully registered" % dsn)
-            METRICS.counter(name="addnewdataset.done").inc()
-            METRICS.counter(name="addnewfile.done").inc(nbfiles)
-            METRICS.timer(name='datasetinjection').observe(stopwatch.elapsed)
-        else:
-            logger(logging.INFO, "Error uploading files")
-        for physical_fname in physical_fnames:
-            remove(physical_fname)
-        rmdir(tmpdir)
+        try:
+            upload_client = UploadClient(client)
+            ret = upload_client.upload(files)
+            if ret == 0:
+                logger(logging.INFO, f"{dsn} successfully registered on {rse}")
+                METRICS.counter(name="addnewdataset.done").inc()
+                METRICS.counter(name="addnewfile.done").inc(nbfiles)
+                METRICS.timer(name='datasetinjection').observe(stopwatch.elapsed)
+                successes.append(rse)
+            else:
+                logger(logging.INFO, "Error uploading files")
+                failures.append(rse)
+        except Exception as error:
+            logger(logging.ERROR, f"Error uploading files on {rse}: {str(error)}")
+            failures.append(rse)
+        finally:
+            for physical_fname in physical_fnames:
+                remove(physical_fname)
+            rmdir(tmpdir)
+    logger(
+        logging.INFO,
+        f"It took {cycle_stopwatch.elapsed} seconds to upload datasets on {len(successes)} RSEs: {str(successes)}",
+    )
     logger(
         logging.INFO,
-        "It took %f seconds to upload one dataset on %s",
-        cycle_stopwatch.elapsed,
-        str(rses),
+        f"Datasets could not be uploaded on {len(failures)} RSEs: {str(failures)}",
```

Comment:
Wouldn't it be better if there is always only one logging message in the format:
`"It took {cycle_stopwatch.elapsed} seconds to successfully upload {len(success)} out of {len(success)+len(failures)}. Successful RSEs : {str(successes)}, failed RSEs : {str(failures)}"` ?

The reason being that in case of partial success (5 successful uploads over 6) it may be weird for the user to read two different logs saying operation was successful and then operation was not successful. Just my 2cents.